tests/test_dev_server.py

```diff
@@ -54,3 +56,19 @@ def test_my_dev_server() -> None:
     assert dev_serv.port == 8000
     assert dev_serv.address == "localhost"
     assert issubclass(dev_serv.my_handler, MyLambdaDevHandler)
+
+
+def test_server_can_run_in_background() -> None:
+    http_process = MyDevServer(HelloWorld, port=9999)
+    http_process.start()
+    url = "http://localhost:9999/"
+
+    request = urllib.request.Request(url, method="GET")
+    try:
+        with urllib.request.urlopen(request) as response:
+            assert response.status == 200
+            assert json.loads(response.read().decode()) == {"message": "HelloWorld"}
+    except Exception as err:
+        raise err
```

Comment:
You don't have to catch an Exception if the only thing you want to do is to raise it again 😉 

`finally` will work anyway.